diff -Naur compiz~/include/core/CMakeLists.txt compiz/include/core/CMakeLists.txt
--- compiz~/include/core/CMakeLists.txt   2012-06-16 18:03:19.413720979 +0200
+++ compiz/include/core/CMakeLists.txt   2012-06-16 18:15:27.468619040 +0200
@@ -23,4 +23,4 @@
     abiversion.h
 )
 
-install (FILES ${_headers} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${_headers} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
diff -Naur compiz~/src/logmessage/CMakeLists.txt compiz/src/logmessage/CMakeLists.txt
--- compiz~/src/logmessage/CMakeLists.txt   2012-06-16 18:03:19.466720317 +0200
+++ compiz/src/logmessage/CMakeLists.txt   2012-06-16 18:15:27.465619076 +0200
@@ -30,4 +30,4 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
diff -Naur compiz~/src/pluginclasshandler/CMakeLists.txt compiz/src/pluginclasshandler/CMakeLists.txt
--- compiz~/src/pluginclasshandler/CMakeLists.txt   2012-06-16 18:03:19.468720292 +0200
+++ compiz/src/pluginclasshandler/CMakeLists.txt   2012-06-16 18:15:27.465619076 +0200
@@ -43,4 +43,4 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
diff -Naur compiz~/src/point/CMakeLists.txt compiz/src/point/CMakeLists.txt
--- compiz~/src/point/CMakeLists.txt   2012-06-16 18:03:19.470720266 +0200
+++ compiz/src/point/CMakeLists.txt   2012-06-16 18:15:27.464619088 +0200
@@ -52,7 +52,7 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
 
 TARGET_LINK_LIBRARIES(
   compiz_point
diff -Naur compiz~/src/rect/CMakeLists.txt compiz/src/rect/CMakeLists.txt
--- compiz~/src/rect/CMakeLists.txt   2012-06-16 18:03:19.472720241 +0200
+++ compiz/src/rect/CMakeLists.txt   2012-06-16 18:15:27.464619088 +0200
@@ -50,7 +50,7 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
 
 TARGET_LINK_LIBRARIES(
   compiz_rect
diff -Naur compiz~/src/string/CMakeLists.txt compiz/src/string/CMakeLists.txt
--- compiz~/src/string/CMakeLists.txt   2012-06-16 18:03:19.475720205 +0200
+++ compiz/src/string/CMakeLists.txt   2012-06-16 18:15:27.464619088 +0200
@@ -28,4 +28,4 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
diff -Naur compiz~/src/timer/CMakeLists.txt compiz/src/timer/CMakeLists.txt
--- compiz~/src/timer/CMakeLists.txt   2012-06-16 18:03:19.475720205 +0200
+++ compiz/src/timer/CMakeLists.txt   2012-06-16 18:15:27.464619088 +0200
@@ -55,7 +55,7 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
 
 TARGET_LINK_LIBRARIES(
   compiz_timer
diff -Naur compiz~/src/window/constrainment/CMakeLists.txt compiz/src/window/constrainment/CMakeLists.txt
--- compiz~/src/window/constrainment/CMakeLists.txt   2012-06-16 18:03:19.478720166 +0200
+++ compiz/src/window/constrainment/CMakeLists.txt   2012-06-16 18:15:27.463619100 +0200
@@ -53,7 +53,7 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
 
 TARGET_LINK_LIBRARIES(
   compiz_window_constrainment
diff -Naur compiz~/src/window/extents/CMakeLists.txt compiz/src/window/extents/CMakeLists.txt
--- compiz~/src/window/extents/CMakeLists.txt   2012-06-16 18:03:19.479720153 +0200
+++ compiz/src/window/extents/CMakeLists.txt   2012-06-16 18:15:27.463619100 +0200
@@ -51,7 +51,7 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
 
 TARGET_LINK_LIBRARIES(
   compiz_window_extents
diff -Naur compiz~/src/window/geometry/CMakeLists.txt compiz/src/window/geometry/CMakeLists.txt
--- compiz~/src/window/geometry/CMakeLists.txt   2012-06-16 18:03:19.481720129 +0200
+++ compiz/src/window/geometry/CMakeLists.txt   2012-06-16 18:15:27.463619100 +0200
@@ -51,7 +51,7 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
 
 TARGET_LINK_LIBRARIES(
   compiz_window_geometry
diff -Naur compiz~/src/window/geometry-saver/CMakeLists.txt compiz/src/window/geometry-saver/CMakeLists.txt
--- compiz~/src/window/geometry-saver/CMakeLists.txt   2012-06-16 18:03:19.480720141 +0200
+++ compiz/src/window/geometry-saver/CMakeLists.txt   2012-06-16 18:15:27.463619100 +0200
@@ -53,7 +53,7 @@
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
 )
 
-install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_CORE_INCLUDE_DIR})
+install (FILES ${PUBLIC_HEADERS} DESTINATION ${COMPIZ_DESTDIR}${COMPIZ_CORE_INCLUDE_DIR})
 
 TARGET_LINK_LIBRARIES (
   compiz_window_geometry_saver 
