 diff --git a/cmake/FindCompiz.cmake b/cmake/FindCompiz.cmake
index b19bcf2..5f101af 100644
--- a/cmake/FindCompiz.cmake
+++ b/cmake/FindCompiz.cmake
@@ -53,23 +53,6 @@ if (NOT _COMPIZ_INTERNAL)
 	    endif ()
 	endif ()
 
-	# add install prefix to pkgconfig search path if needed
-        string (REGEX REPLACE "([\\+\\(\\)\\^\\\$\\.\\-\\*\\?\\|])" "\\\\\\1" PKGCONFIG_REGEX ${CMAKE_INSTALL_PREFIX})
-	set (PKGCONFIG_REGEX ".*${PKGCONFIG_REGEX}/lib/pkgconfig:${PKGCONFIG_REGEX}/share/pkgconfig.*")
-
-	if (NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-
-	    if (NOT "$ENV{PKG_CONFIG_PATH}" MATCHES "${PKGCONFIG_REGEX}")
-		if ("" STREQUAL "$ENV{PKG_CONFIG_PATH}")
-		    set (ENV{PKG_CONFIG_PATH} "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig:${CMAKE_INSTALL_PREFIX}/share/pkgconfig")
-	    	else ()
-		    set (ENV{PKG_CONFIG_PATH}
-		         "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig:${CMAKE_INSTALL_PREFIX}/share/pkgconfig:$ENV{PKG_CONFIG_PATH}")
-		endif ()
-	    endif ()
-
-	endif (NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-
 	# look for compiz
 	pkg_check_modules (COMPIZ ${_req} "compiz${_comp_ver}")
 
diff --git a/cmake/base.cmake b/cmake/base.cmake
index 7731017..db4683c 100644
--- a/cmake/base.cmake
+++ b/cmake/base.cmake
@@ -37,16 +37,3 @@ function (_check_compiz_cmake_macro)
 	${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake ${CMAKE_ROOT}/Modules
     )
 endfunction ()
-
-# add install prefix to pkgconfig search path if needed
-string (REGEX REPLACE "([\\+\\(\\)\\^\\\$\\.\\-\\*\\?\\|])" "\\\\\\1" PKGCONFIG_REGEX ${CMAKE_INSTALL_PREFIX})
-set (PKGCONFIG_REGEX ".*${PKGCONFIG_REGEX}/lib/pkgconfig:${PKGCONFIG_REGEX}/share/pkgconfig.*")
-
-if (NOT "$ENV{PKG_CONFIG_PATH}" MATCHES "${PKGCONFIG_REGEX}")
-    if ("" STREQUAL "$ENV{PKG_CONFIG_PATH}")
-	set (ENV{PKG_CONFIG_PATH} "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig:${CMAKE_INSTALL_PREFIX}/share/pkgconfig")
-    else ()
-	set (ENV{PKG_CONFIG_PATH}
-	    "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig:${CMAKE_INSTALL_PREFIX}/share/pkgconfig:$ENV{PKG_CONFIG_PATH}")
-    endif ()
-endif ()
diff --git a/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake b/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake
index ccc63db..9ddc870 100644
--- a/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake
+++ b/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake
@@ -51,19 +51,6 @@ if (NOT _COMPIZCONFIG_INTERNAL)
 	    endif ()
 	endif ()
 
-	# add install prefix to pkgconfig search path if needed
-        string (REGEX REPLACE "([\\+\\(\\)\\^\\\$\\.\\-\\*\\?\\|])" "\\\\\\1" PKGCONFIG_REGEX ${CMAKE_INSTALL_PREFIX})
-	set (PKGCONFIG_REGEX ".*${PKGCONFIG_REGEX}/lib/pkgconfig:${PKGCONFIG_REGEX}/share/pkgconfig.*")
-
-	if (NOT "$ENV{PKG_CONFIG_PATH}" MATCHES "${PKGCONFIG_REGEX}")
-	    if ("" STREQUAL "$ENV{PKG_CONFIG_PATH}")
-		set (ENV{PKG_CONFIG_PATH} "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig:${CMAKE_INSTALL_PREFIX}/share/pkgconfig")
-	    else ()
-		set (ENV{PKG_CONFIG_PATH}
-		    "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig:${CMAKE_INSTALL_PREFIX}/share/pkgconfig:$ENV{PKG_CONFIG_PATH}")
-	    endif ()
-	endif ()
-
 	# look for compiz
 	pkg_check_modules (COMPIZCONFIG ${_req} "libcompizconfig")
