diff --git a/src/dxvk/dxvk_interfaces.h b/src/dxvk/dxvk_interfaces.h
index d4e5f3c..1a733ce 100644
--- a/src/dxvk/dxvk_interfaces.h
+++ b/src/dxvk/dxvk_interfaces.h
@@ -6,8 +6,13 @@
 #pragma GCC diagnostic ignored "-Wnon-virtual-dtor"
 #endif // __GNUC__
 
-#define DXVK_DEFINE_GUID(iface) \
-  template<> inline GUID const& __mingw_uuidof<iface> () { return iface::guid; }
+#ifdef __WINE__
+  #define DXVK_DEFINE_GUID(iface) \
+    template<> inline GUID const& __wine_uuidof<iface> () { return iface::guid; }
+#else
+  #define DXVK_DEFINE_GUID(iface) \
+    template<> inline GUID const& __mingw_uuidof<iface> () { return iface::guid; }
+#endif
 
 enum D3D11_VK_EXTENSION : uint32_t {
   D3D11_VK_EXT_MULTI_DRAW_INDIRECT        = 0,
diff --git a/src/nvapi_private.h b/src/nvapi_private.h
index e79370e..910d768 100644
--- a/src/nvapi_private.h
+++ b/src/nvapi_private.h
@@ -47,6 +47,23 @@ typedef enum {
 #define NV_DECLARE_HANDLE(name) struct name##__ { int unused; }; typedef struct name##__ *name
 NV_DECLARE_HANDLE(NVDX_ObjectHandle);
 
-#define NVAPI_INTERFACE __declspec(dllexport) NvAPI_Status __cdecl
-#define NVAPI_QUERYINTERFACE __declspec(dllexport) void* __cdecl 
+#if defined(__WINE__)
+  // GCC: -std options disable certain keywords
+  // https://gcc.gnu.org/onlinedocs/gcc/Alternate-Keywords.html
+  #if !defined(typeof)
+    #define typeof __typeof
+  #endif
+
+  template<typename T>
+  void **IID_PPV_ARGS_Helper (T **pp) { return reinterpret_cast<void **> (pp); }
+
+  #define IID_PPV_ARGS(ppType) __uuidof (**(ppType)), IID_PPV_ARGS_Helper (ppType)
+
+  #define DLLEXPORT __attribute__((visibility("default")))
+#else
+  #define DLLEXPORT __declspec(dllexport)
+#endif
+
+#define NVAPI_INTERFACE NvAPI_Status __cdecl
+#define NVAPI_QUERYINTERFACE DLLEXPORT void* __cdecl
 #define NVAPI_CAST reinterpret_cast<void*>
