--- a/src/d3d9/meson.build
+++ b/src/d3d9/meson.build
@@ -36,6 +36,7 @@ d3d9_dll = shared_library('d3d9'+dll_ext, d3d9_src, glsl_generator.process(d3d9_
   dependencies        : [ dxso_dep, dxvk_dep ],
   include_directories : dxvk_include_path,
   install             : true,
+  install_dir         : get_option('libdir') / 'd3d9',
   objects             : not dxvk_msvc ? 'd3d9'+def_spec_ext : [],
   vs_module_defs      : 'd3d9'+def_spec_ext,
   override_options    : ['cpp_std='+dxvk_cpp_std])
