--- a/build/pgo/profileserver.py	2019-09-30 20:58:31.169527743 +0700
+++ b/build/pgo/profileserver.py	2019-09-30 21:02:07.964185254 +0700
@@ -137,7 +137,6 @@
                                env=env,
                                process_args=process_args)
         runner.start(debug_args=debug_args, interactive=interactive)
-        ret = runner.wait()
         httpd.stop()
         if ret:
             print("Firefox exited with code %d during profiling" % ret)
