--- a/proton
+++ b/proton
@@ -333,6 +333,7 @@
                     try_copy(steamdir + "/legacycompat/" + f, dstfile)
 
             #copy openvr files into place
+            '''
             dst = self.prefix_dir + "/drive_c/vrclient/bin/"
             makedirs(dst)
             try_copy(g_proton.lib_dir + "wine/fakedlls/vrclient.dll", dst)
             try_copy(g_proton.lib64_dir + "wine/fakedlls/vrclient_x64.dll", dst)

+            '''
 
             if "wined3d" in g_session.compat_config:
                 dxvkfiles = []
