--- a/backends/documentinternal.h	2019-06-09 09:03:01.000000000 +0200
+++ b/backends/documentinternal.h	2021-01-05 21:15:55.723321380 +0100
@@ -33,6 +33,7 @@
 #include <map>
 #include <memory>
 #include <string>
+#include <limits>
 
 class DocumentTermList;
 class DocumentValueList;

--- a/api/terminfo.cc     2018-10-31 09:03:01.000000000 +0100
+++ b/api/terminfo.cc    2021-01-05 21:24:31.806644502 +0100
@@ -25,6 +25,7 @@
 #include "omassert.h"
 
 #include <algorithm>
+#include <limits>
 
 using namespace std;
 
